decidim-budgets/app/controllers/decidim/budgets/orders_controller.rb

```diff
@@ -7,8 +7,6 @@ class OrdersController < Decidim::Budgets::ApplicationController
       include NeedsCurrentOrder
 
       def checkout
-        authorize_action! "vote"
```

Comment:
You might need to add something similar to `enforce_permission_to :vote, :project, project: project` here 🤔 Otherwise we're no longer checking for permissions in this method...

Reply:
I'd avoid the first solution you propose, as I'd like to double-ensure the user can vote so we avoid any weird hacking (or at least we try to). I'd use any of the two other options, for example refactoring the `can_vote_project?` so that it can receive a `project` as a param, and then you call `can_vote_project?(project)` or `can_vote_project?(order.projects.first)` as you need. 

How does this sound? 😄

Reply:
This would solve the weird double redirect error, I guess?

Reply:
I guess that the double redirect problem is a design issue of the `authorize_action!` method: it doesn't let the calling method know if the action was not authorized to abort the execution of the rest of the method.